chain/ethereum/src/runtime/abi.rs

```diff
@@ -0,0 +1,290 @@
+use graph::prelude::BigInt;
+use graph::runtime::{asc_get, asc_new, AscPtr, DeterministicHostError, FromAscObj, ToAscObj};
+use graph::runtime::{AscHeap, AscType};
+use graph_runtime_derive::AscType;
+use graph_runtime_wasm::asc_abi::class::{
+    Array, AscAddress, AscBigInt, AscEnum, AscH160, AscString, EthereumValueKind, Uint8Array,
+};
+
+use crate::trigger::{
+    EthereumBlockData, EthereumCallData, EthereumEventData, EthereumTransactionData,
+};
+
+use super::runtime_adapter::UnresolvedContractCall;
+
+type AscH256 = Uint8Array;
+type AscLogParamArray = Array<AscPtr<AscLogParam>>;
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscUnresolvedContractCall_0_0_4 {
```

Comment:
What's the significance of the `0_0_4` suffix?

Reply:
Subgraph's apiVersion